### ISSUE TITLE
[Feature]: Support for SQLite3

### ISSUE THREAD

**Issue by Nikhil-Nandagopal** [appsmithorg/appsmith#8780]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

A user has requested for an SQLite3 DB integration

### Why should this be worked on?

SQLite is a widely used SQL datasource and should be very similar to our exisiting datasources

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_5z7f0)

**Comment by szabolcsmaj** [comment]
I second this. SQLite is used for many micro/small projects and the biggest issue comes when one has to "throw" a quick UI over the DB. Is this on the roadmap for the future?

**Comment by Nikhil-Nandagopal** [comment]
Hello! We don't currently have this on our roadmap since we haven't received too many requests for it yet. We'd be very open to a community contribution on this datasource!

**Comment by supermx1** [comment]
There are a lot of developers that need this. Sqlite is a really used database

**Comment by arjunsriva-woven** [comment]
Agree, we'd love to have this for some of our internal tools for non-developers too

**Comment by nickdex** [comment]
I too use sqlite for many of my projects and would love to see appsmith support it.

**Comment by viruha** [comment]
Vote to have sqlite!!

**Comment by bobbintb** [comment]
+1

**Comment by Uphorium** [comment]
+1 It has some great uses.

**Comment by pasalgo-project** [comment]
+1 Will make it the premier sqlite internal tool

**Comment by dontmint** [comment]
+1 please consider this database

**Comment by FJEagle** [comment]
+1, sqlite or  built-in database like ToolJet Database

**Comment by Nikhil-Nandagopal** [comment]
@FJEagle any reason you aren't ok with using the supabase.com free tier?

**Comment by QuanVan95** [comment]
+1 please consider this database

**Comment by christian-korneck** [comment]
this was also one of the first things I was looking for but couldn't find.

**Comment by Quetzal-coalt** [comment]
+1 Please i really love appsmith and i have a lot of small projet with sqlite, i really want to mutulize all of my app with appsmith

**Comment by keerio** [comment]
+1

**Comment by reignmaker** [comment]
Any news on this?

**Comment by ivishalgandhi** [comment]
+1

**Comment by MitPitt** [comment]
+1

**Comment by MitPitt** [comment]
SQLite is one of the most popular databases, on par with Postgres. Not supporting it is baffling.

**Comment by victor9000** [comment]
> SQLite is one of the most popular databases, on par with Postgres. Not supporting it is baffling.

It would be great for users that self-host, but maybe that's the issue.

**Comment by shanid544** [comment]
@NilanshBansal. Shall I work on this?

**Comment by shanid544** [comment]
@NilanshBansal Sqlite support is when appsmith hosted in local? or do we need to think taking leverage of web apps?

**Comment by NilanshBansal** [comment]
@shanid544 this functionality should work for both appsmith cloud as well as self hosted.

**Comment by shanid544** [comment]
@NilanshBansal I will start with self hosted support

**Comment by shanid544** [comment]
@NilanshBansal To integrate SQLite with Appsmith, I have explored several approaches and decided to leverage existing cloud solutions. During my research, I came across [SQLiteCloud](https://sqlitecloud.io/) and am considering utilizing its WebLite service for seamless integration. I found the documentation for WebLite at [SQLiteCloud WebLite Docs](https://docs.sqlitecloud.io/docs/weblite), which provides useful insights. Should I proceed with this approach, or would you have any other suggestions?

**Comment by NilanshBansal** [comment]
Thanks @shanid544 for your exploration. As discussed can you please setup a call with me using [this](https://calendly.com/nilansh/30min) link.

**Comment by kelvinauta** [comment]
I installed AppSmith and uninstalled it immediately upon seeing that there was no SQLite support.
It's been five years since this issue was opened — I don't think there's any hope.
